immunopepper/traversal.py

```diff
@@ -330,44 +311,31 @@ def get_and_write_peptide_and_kmer(peptide_set=None, kmer_dict=None,
-                                       segment_expr=segment_expr_meta,
                                        kmer_type=kmer_type
                                        ), sep = '\t'))
                     variant_id += 1
-                    output_peptide = OutputPeptide(output_id= new_output_id,
+                    output_peptide = OutputPeptide(output_id=new_output_id,
                                                     peptide=peptide.mut[pep_idx],
                                                     exons_coor=modi_coord,
-                                                    junction_expr=edge_expr,
-                                                    junction_annotated=vertex_tuple_anno_flag,
+                                                    strand=gene.strand,
                                                     read_frame_annotated=vertex_pair.read_frame.annotated_RF)
 
                     ### kmers
-                    if cross_graph_expr: #generate kmer x sample expression matrix for all samples in graph
-                        create_output_kmer_cross_samples(output_peptide, kmer[0], expr_list, graph_output_samples_ids,
-                                                         kmer_matrix_edge, kmer_matrix_segm, kmer_database,
-                                                         graph_samples, filepointer,
-                                                         out_dir=out_dir, verbose=verbose_save)
-                        # Only one kmer length supported for this mode
+                    if '2-exons' in kmer_type:
+                        for kmer_length in kmer: # generate sample kmers for each vertex pair and each kmer_length
+                            create_kmer(output_peptide, kmer_length, gene_kmer_coord, kmer_database)
 
-                    else:
-                        # generate sample kmers for each vertex pair and each kmer_length
-                        if '2-exons' in kmer_type:
-                            for kmer_length in kmer:
-                                kmer_dict[kmer_length].update(create_output_kmer(output_peptide, kmer_length,
-                                                                              expr_list, kmer_database))
-                        # generate sample kmers for each vertex triplets, only for the kmer_lengths that require it
-                        else:
-                            kmer_length = int(kmer_type.split('_')[-1].split('-')[0])
-                            kmer_dict[kmer_length].update(create_output_kmer(output_peptide, kmer_length,
-                                                                          expr_list, kmer_database))
+                    else: # if two exons do not allow to reach the desired kmer length, add 3rd exon
+                        kmer_length = int(kmer_type.split('_')[-1].split('-')[0])
+                        create_kmer(output_peptide, kmer_length, gene_kmer_coord, kmer_database)
```

Comment:
Yes we do not allow multiple kmer length. This does the right thing but is not elegant because kmer_length becomes, a simple string and kmer_type denomination can be simplified